### ISSUE TITLE
AttributeError: 'DictionaryObject' object has no attribute 'indirect_reference'

### ISSUE THREAD

**Issue by jainharsh97** [py-pdf/pypdf#1728]
I'm trying to merge a few PDF files using this library into one single file by just appending them into a new document one-by-one. However, some of the input files seem to run into the error described in the title above when trying to call `append(fileobj=open(...))`

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-13.1-x86_64-i386-64bit

$ python -c "import pypdf;print(pypdf.__version__)"
3.5.2
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
output_pdf_stream = PdfWriter()
output_pdf_stream.append(fileobj=input, pages=(3, 5))
```
## Full example 
```
from pypdf import errors, PdfWriter

def main():
    output_pdf_stream = PdfWriter()
    input = open("inputFile.pdf", "rb")
    output_pdf_stream.append(fileobj=input, pages=(3, 5))

    output_pdf_name = "output.pdf"
    # does not reach here
    # create output pdf file
    try:
        output_pdf_file = open(output_pdf_name, "wb")
        output_pdf_stream.write(output_pdf_file)
    finally:
        output_pdf_file.close()

    output_pdf_stream.close()

    print("%s successfully created." % output_pdf_name)


if __name__ == "__main__":
    main()
```
[inputFile.pdf](https://github.com/py-pdf/pypdf/files/11015242/inputFile.pdf)

## Traceback

This is the complete Traceback I see:

```
Traceback (most recent call last):
  File "/Users/harshjain/mt/picasso-document-converter-filestack/pypdftest.py", line 13, in <module>
    merger.append(fileobj=input2, pages=(3, 5))
  File "/usr/local/lib/python3.10/site-packages/pypdf/_writer.py", line 2535, in append
    self.merge(
  File "/usr/local/lib/python3.10/site-packages/pypdf/_utils.py", line 442, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/pypdf/_writer.py", line 2658, in merge
    lst = self._insert_filtered_annotations(
  File "/usr/local/lib/python3.10/site-packages/pypdf/_writer.py", line 2845, in _insert_filtered_annotations
    outlist.append(anc.indirect_reference)
AttributeError: 'DictionaryObject' object has no attribute 'indirect_reference'
```

**Comment by jainharsh97** [comment]
Just a side note, the above error occurs when using `PdfWriter` and is solved by using `PdfMerger` class

**Comment by MartinThoma** [comment]
The issue was fixed in #1729 which I just merged to main. It will be released latest on 26.03.2023 to PyPI with `pypdf > 3.6.0`

**Comment by MartinThoma** [comment]
@jainharsh97 Thank you for the report :pray: If you want, I can add you as a contributor: https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html
